openlibrary/coverstore/warc.py

```diff
@@ -146,12 +146,14 @@ class WARCHeader:
 
     >>> WARCHeader("WARC/0.10", 10, "resource", "subject_uri", "20080808080808", "record_42", "image/jpeg", {'hello': 'world'})
     <header: 'WARC/0.10 10 resource subject_uri 20080808080808 record_42 image/jpeg\r\nhello: world\r\n\r\n'>
+    >>> WARCHeader("WARC/0.10", b"10", "resource", "subject_uri", "20080808080808", "record_42", "image/jpeg", {'hello': 'world'})
+    <header: 'WARC/0.10 10 resource subject_uri 20080808080808 record_42 image/jpeg\r\nhello: world\r\n\r\n'>
     """
     def __init__(self, warc_id,
             data_length, record_type, subject_uri,
             creation_date, record_id, content_type, headers):
         self.warc_id = warc_id
-        self.data_length = str(data_length)
+        self.data_length = str(int(data_length))  # deal with bytes, int, and str
```

Comment:
Fair.

Reply:
Hah :) I guess the answer to the dumb question is that my preference for laziness would do the job :)

I agree that it might make sense to create that option of taking int vs. bytes vs. str there. I would prefer to do that in a separate PR than one focused on Python 3-ification, only because I have personally tried adding features/flexibility to codebases while Python 3-ifying them, then found myself in a project that ran way longer than I wanted.